### PR TITLE
UI/15529836 advanced page

### PR DESCRIPTION
**Pull request by CzarekDryl**
Added advanced user page.

https://tw.auroracreation.com/app/tasks/15529836
https://tw.auroracreation.com/app/tasks/15529855

image

### CODE REVIEW

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import Navigation from '~v5/common/Navigation';
+import TwoColumns from '~v5/frame/TwoColumns';
+import Spinner from '~v5/shared/Spinner';
+import { useUserAdvancedPage } from './hooks';
+import { HookForm } from '~shared/Fields';
+import Toast from '~shared/Extensions/Toast';
+import SettingsRow from '../SettingsRow/SettingsRow';
+import SettingsInputRow from '../SettingsInputRow';
+
+const displayName = 'v5.pages.UserAdvancedPage.partials.UserAdvancedSettings';
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx:15]: This is whole page component so I will leave it there

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    metatransasctionsDefault,
+    rpcValidationSchema,
+    metatransactionsValidationSchema,
+    decentralizedModeEnabled,
+  } = useUserAdvancedPage();
+
+  return (
+    <Spinner loadingText={{ id: 'loading.userAdvancedPage' }}>
+      <TwoColumns aside={<Navigation pageName="profile" />}>
+        <HookForm
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx:33]: can you move this form to separate compoent?

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                        ? 'advancedSettings.fees.toast.description.true'
+                        : 'advancedSettings.fees.toast.description.false',
+                    }}
+                  />,
+                );
+              }}
+            />
+          </div>
+        </HookForm>
+        <HookForm
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx:66]: can you move this form to separate component?

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            [SlotKey.CustomRPC]: customRpc,
+          }}
+        >
+          <div className="border-b border-gray-200">
+            <SettingsRow
+              title={{ id: 'advancedSettings.rpc.title' }}
+              description={{ id: 'advancedSettings.rpc.description' }}
+              tooltipMessage={{ id: 'advancedSettings.rpc.tooltip' }}
+              id={SlotKey.DecentralizedMode}
+              onChange={(value) => {
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx:79]: can you move logic to separate function onChange and pass it here?

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            {formatMessage({ id: 'userAdvancedPage.title' })}
+          </h4>
+          <div className="border-b border-gray-200">
+            <SettingsRow
+              title={{ id: 'advancedSettings.fees.title' }}
+              description={{ id: 'advancedSettings.fees.description' }}
+              tooltipMessage={{ id: 'advancedSettings.fees.tooltip' }}
+              id={SlotKey.Metatransactions}
+              onChange={(value) => {
+                handleSubmit({ [SlotKey.Metatransactions]: value });
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/UserAdvancedSettings.tsx:50]: can you move logic to separate function onChange and pass it here?

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  const metatransactionsAvailable = canUseMetatransactions();
+  const {
+    settings: {
+      metatransactions: metatransactionsSetting,
+      decentralizedModeEnabled,
+      customRpc,
+    },
+    setSettingsKey,
+  } = useUserSettings();
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/hooks.ts:59]: add empty line

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/hooks.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  const [isInputVisible, setIsInputVisible] = useState(
+    decentralizedModeEnabled,
+  );
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/hooks.ts:62]: add empty line

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/types.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export interface AdvancedSettingsFields {
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/types.ts:1]: please generete this interface using AdvancedSettingsFields form src/components/frame/v5/pages/UserAdvancedPage/types.ts file and Omit.

File: src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/types.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+  metatransactions: boolean;
+  decentralizedModeEnabled: boolean;
+  customRpc: string;
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/UserAdvancedSettings/types.ts:4]: customRpc: string; and customRPC: string; shouldn't be called the same?

File: src/components/shared/Fields/Form/HookForm.tsx [hunk 10]
```diff
@@ -76,7 +76,7 @@ const HookForm = <FormData extends FieldValues>({
     <FormProvider {...formHelpers}>
       <form
         onSubmit={handleSubmit(
-          (data, e) => onSubmit(data, formHelpers, e),
+          (data, e) => onSubmit && onSubmit(data, formHelpers, e),
```
Comment (krawczykkatarzyna) [inline src/components/shared/Fields/Form/HookForm.tsx:79]: just `onSubmit?.(data, formHelpers, e)`

File: src/components/v5/common/SettingsInputRow/SettingsInputRow.tsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+                  isValid && rpcValue
+                    ? 'advancedSettings.rpc.successful'
+                    : undefined
+                }
+              />
+              <Button
+                mode="primarySolid"
+                className="mt-6 ml-auto"
+                disabled={!isValid || !rpcValue || !!error || !isDirty}
+                onClick={() => {
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/SettingsInputRow/SettingsInputRow.tsx:80]: can you move the logic to separate function and pass it here?

**Review by krawczykkatarzyna — APPROVED** [review]

File: src/components/v5/common/SettingsRow/SettingsRow.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
+import React, { FC } from 'react';
+import { useIntl } from 'react-intl';
+
+import { SettingsRowProps } from './types';
+import Switch from '~v5/common/Fields/Switch';
+import Button from '~v5/shared/Button';
+import Icon from '~shared/Icon';
+import Tooltip from '~shared/Extensions/Tooltip';
+
+const displayName = 'v5.pages.UserAdvancedPage.partials.SettingsRow';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/SettingsRow/SettingsRow.tsx:10]: Please, correct the displayName

File: src/components/v5/common/SettingsInputRow/SettingsInputRow.tsx [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+import { toast } from 'react-toastify';
+
+import { SettingsInputRowProps } from './types';
+import { SlotKey } from '~hooks';
+import Button from '~v5/shared/Button';
+import Input from '~v5/common/Fields/Input';
+import Toast from '~shared/Extensions/Toast';
+import { useSettingsInputRow } from './hooks';
+
+const displayName = 'v5.pages.UserAdvancedPage.partials.SettingsInputRow';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/SettingsInputRow/SettingsInputRow.tsx:12]: Please, correct the displayName

File: src/components/frame/v5/pages/UserAdvancedPage/partials/RpcForm/RpcForm.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/RpcForm/RpcForm.tsx:1]: Shouldn't be in components, not pages directory?

File: src/components/frame/v5/pages/UserAdvancedPage/partials/FeesForm/FeesForm.tsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/v5/pages/UserAdvancedPage/partials/FeesForm/FeesForm.tsx:1]: Shouldn't be in components, not pages directory?

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

**Pull request merged**